SymPEP-0001.md

```diff
@@ -0,0 +1,241 @@
+completed. When the reference implementation is complete and incorporated into
+the main source code repository, the status will be changed to **Final**.
+
+A SymPEP can also be assigned status **Deferred**. The SymPEP author or a core
+developer can assign the SymPEP this status when no progress is being made on
+the SymPEP.
+
+A SymPEP can also be **Rejected**. Perhaps after all is said and done it was
+not a good idea. It is still important to have a record of this fact. The
+**Withdrawn** status is similar, it means that the SymPEP author themselves has
+decided that the SymPEP is actually a bad idea or has accepted that a competing
+proposal is a better alternative.
+
+SymPEPs can also be **Superseded** by a different SymPEP, rendering the
+original obsolete.
+
+Process SymPEPs may also have a status of **Active** if they are never meant to
+be completed, e.g. SymPEP 1 (this SymPEP).
+
+### Merging the SymPEP Document Pull Request
+
+Whenever a SymPEP moves from the **Draft** status to one of the other above
+statuses, the header should be updated, and the corresponding pull request
+should be merged. This way the document lives inside of the SymPEP repository
+proper. SymPEPs that are merged are not set in stone, and may be updated by
+future pull requests (although SymPEPs that are **Accepted** should generally
+not be significantly modified once they have reached that status). The purpose
+of merging is simply to make the SymPEP visible in the repository, even if it
+isn't yet accepted or rejected. SymPEP discussions that have stalled should
+also be merged, so that the SymPEP becomes visible in the SymPEP repository
+proper—again, discussion may be picked up again with a new pull request.
+
+## Accepting a SymPEP
+
+Once a SymPEP is Accepted by consensus of all interested contributors, an email
+should be sent to the [SymPy mailing
+list](http://groups.google.com/group/sympy) with a subject like:
+
+    Proposal to accept SymPEP #<number>: <title>
+
+In the body of your email, you should:
+
+- link to the latest version of the SymPEP,
+
+- briefly describe any major points of contention and how they were resolved,
+
+- include a sentence like: “If there are no substantive objections within 7
+  days from this email, then the SymPEP will be accepted; see SymPEP 1 for more
+  details.”
+
+After you send the email, add the the email thread to the Discussion section of
+the SymPEP, so that people can find it later.
+
+Generally the SymPEP author will be the one to send this email, but anyone can
+do it. The important thing is to make sure that everyone knows when a SymPEP is
+on the verge of acceptance, and give them a final chance to respond. If there's
+some special reason to extend this final comment period beyond 7 days, then
+that's fine, just say so in the email. It shouldn't be less than 7 days,
+because sometimes people are traveling or similar and need some time to
+respond.
```

Comment:
I think that 7 days is quite short for many things. I can see that the intention here is just to allow a little time after people discussing something on GitHub somewhere have agreed for a broader audience to review, comment or object so in practice the discussion will have taken more than 7 days. I think that there should be a mailing list announcement at the very start of creating a SymPEP proposal though so that people can begin participating at that point or have more time to ask questions, make comments, etc. Above it says that discussions should usually start on the mailing list and perhaps it should just be clear that if that discussion turns into a SymPEP proposal then a mailing list announcement should clarify if discussions are going to take place elsewhere.

Reply:
Perhaps it would be useful just to note a vague timeline like:

1. Initiate discussion in mailing list, issue or PR - someone might suggest a SymPEP.
2. If discussion did not start on the mailing list then move to the mailing list (with cross-reference links) to suggest a SymPEP on the mailing list.
3. Initial discussion on mailing list indicates a SymPEP is appropriate.
4. Create a PR for the SymPEP and announce it on the mailing list along with a suggestion of primary forum for detailed discussion.
5. Detailed discussion takes place in the designated forum until a "consensus" of those participating is reached.
6. Announce the final draft of SymPEP on the mailing list and then it can be 7 days or perhaps longer if indicated for anyone to comment.
7. Mark the SymPEP accepted.

Obviously the above describes a smooth process in which everyone agrees about everything but at any of the above stages there might be alternative decisions like not having a SymPEP, rejecting it, pursuing a different idea etc.